base/timing.jl

```diff
@@ -198,20 +202,34 @@ julia> @time begin
 ```
 """
 macro time(ex)
+    if Threads.threadid() == 1 && latest_time_cumucomptime[] == 0 # catch the first time `@time` is interpreted
+        latest_time_cumucomptime[] = cumulative_compile_time_ns_before()
```

Comment:
Macro expansion should not have side effects, so this is not a great way to implement this.